nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-multi/src/main/java/org/nuxeo/ecm/directory/multi/MultiReference.java

```diff
@@ -83,36 +89,30 @@ public void addLinks(List<String> sourceIds, String targetId) throws DirectoryEx
                 }
             }
         }
-        List<String> x = new ArrayList<String>(ids.size());
+        List<String> x = new ArrayList<>(ids.size());
         x.addAll(ids);
         return x;
     }
 
     @Override
     public List<String> getSourceIdsForTarget(final String targetId) throws DirectoryException {
-        return doCollect(new Collector() {
-            @Override
-            public List<String> collect(List<Reference> refs) throws DirectoryException {
-                List<String> sourceIds = new ArrayList<>(1);
-                for (Reference ref : refs) {
-                    sourceIds.addAll(ref.getSourceIdsForTarget(targetId));
-                }
-                return sourceIds;
+        return doCollect(refs -> {
+            List<String> sourceIds = new ArrayList<>(1);
+            for (Reference ref : refs) {
+                sourceIds.addAll(ref.getSourceIdsForTarget(targetId));
             }
+            return sourceIds;
```

Comment:
1 is the default size which is overwhelmingly the common number of sources/targets. It doesn't prevent the array from growing when needed.
Yes streams would be better but let's optimize in later commits, this is mostly just a refactoring.